crates/ursa-network/src/service.rs

```diff
@@ -236,6 +240,45 @@ where
             event_receiver,
             response_channels: Default::default(),
             index_provider,
+            config: config.clone(),
+        }
+    }
+
+    // Make an http node announcement to a ursa tracker. Used for mapping the network pre-consensus
+    pub async fn announce_node(&self) -> Result<String> {
```

Comment:
Perhaps using a another name than `announce_node` too similar to announcing in index-provider

Reply:
Agreed its a little confusing on the terminology. 

Maybe `register_node` might be better?

Reply:
2c8ebc9a0959fad47629f6efb80f6b3898261123